### PR TITLE
补全数组与链表/数组（Array）中C语言部分

### PR DESCRIPTION
**Pull request by Ayuan66**
If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_array_and_linkedlist/array.md [hunk 1]
```diff
@@ -174,7 +177,14 @@ elementAddr = firtstElementAddr + elementLength * elementIndex
 === "C"
 
     ```c title="array.c"
-    
+    /* 随机返回一个数组元素 */
+    int randomAccess(int nums[], int size) {
+        //在区间[0, nums的长度)中随机抽取一个数字
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:182]: Add a space between `//` and the comment.

File: docs/chapter_array_and_linkedlist/array.md [hunk 2]
```diff
@@ -67,7 +67,10 @@ comments: true
 === "C"
 
     ```c title="array.c"
-    
+    /* 初始化数组 */
+     int nums[8] = {1, 2, 4, 8, 16 ,32 ,64}; /* 从ANSI C开始支持这种初始化 即完全初始化 */
+     int arr[5] = {1, 2}; /* 不完全初始化 未初始化的元素会被置0 */
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:72]: Please write as `// 不完全初始化 未初始化的元素会被置 0`.
Add a space between every 中文 and 英文或数字 。

File: docs/chapter_array_and_linkedlist/array.md [hunk 3]
```diff
@@ -293,7 +303,17 @@ elementAddr = firtstElementAddr + elementLength * elementIndex
 === "C"
 
     ```c title="array.c"
-    
+    /*扩展数组长度 */
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:306]: `/* 扩展数组长度 */`

File: docs/chapter_array_and_linkedlist/array.md [hunk 4]
```diff
@@ -67,7 +67,10 @@ comments: true
 === "C"
 
     ```c title="array.c"
-    
+    /* 初始化数组 */
```
Comment (krahets) [inline docs/chapter_array_and_linkedlist/array.md:70]: Please add the source code file `array.c` and test the codes.

**Review by krahets — COMMENTED** [review]
Thanks! Please address the comments and check the code and the comments carefully.

File: docs/chapter_array_and_linkedlist/array.md (hunk 1 above)
Comment (Ayuan66) [reply docs/chapter_array_and_linkedlist/array.md:182]: done

File: docs/chapter_array_and_linkedlist/array.md (hunk 2 above)
Comment (Ayuan66) [reply docs/chapter_array_and_linkedlist/array.md:72]: done

File: docs/chapter_array_and_linkedlist/array.md (hunk 3 above)
Comment (Ayuan66) [reply docs/chapter_array_and_linkedlist/array.md:306]: done

**Comment by krahets** [comment]
This PR is not merged because the source code `array.c` was not provided. If wanting to contribute code, it is recommended to get more familiar with the code and PR formats referring to other PR.

**Pull request closed without merging**